src/web/features.rs

```diff
@@ -38,28 +40,14 @@ pub fn build_features_handler(req: &mut Request) -> IronResult<Response> {
 
     let row = cexpect!(req, rows.get(0));
 
+    let mut features = None;
     let mut default_len = 0;
-    let features = row
-        .get::<'_, usize, Option<Vec<Feature>>>(0)
-        .map(|raw| {
-            raw.into_iter()
-                .filter(|feature| !feature.is_private())
-                .map(|feature| (feature.name.clone(), feature))
-                .collect::<HashMap<String, Feature>>()
-        })
-        .map(|mut feature_map| {
-            let mut features = get_tree_structure_from_default(&mut feature_map);
-            let mut remaining = feature_map
-                .into_iter()
-                .map(|(_, feature)| feature)
-                .collect::<Vec<Feature>>();
-            remaining.sort_by_key(|feature| feature.subfeatures.len());
-
-            default_len = features.len();
-
-            features.extend(remaining.into_iter().rev());
-            features
-        });
+
+    if let Some(raw) = row.get(0) {
+        let result = order_features_and_count_default_len(raw);
+        features = Some(result.0);
+        default_len = result.1;
```

Comment:
https://github.com/rust-lang/rust/pull/71156 would help with this :) 'coming soon to rust versions near you'

Reply:
That will be nice :)